src/hooks/useTodoController.ts

```diff
@@ -0,0 +1,276 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Todo } from '../types/todo/Todo';
+import * as todoService from '../api/todos';
+import { ErrorMessage } from '../enums/errorMessage';
+import { StatusFilter } from '../enums/statusFilter';
+import { getFilteredTodos } from '../utils/getFilteredTodos';
+
+export const useTodoController = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.None,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAddTodoFormFocused, setIsAddTodoFormFocused] = useState(false);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<Todo['id'][]>([]);
+  const [todosToUpdate, setTodosToUpdate] = useState<Todo[]>([]);
+  const [editingTodoId, setEditingTodoId] = useState<Todo['id'] | null>(null);
+  const [isEditTodoFormFocused, setIsEditTodoFormFocused] = useState(false);
+  const [statusFilter, setStatusFilter] = useState<StatusFilter>(
+    StatusFilter.All,
+  );
+
+  useEffect(() => {
+    setIsTodosLoading(true);
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.OnLoad);
+      })
+      .finally(() => setIsTodosLoading(false));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return getFilteredTodos(todos, statusFilter);
+  }, [todos, statusFilter]);
+
+  const {
+    isThereAlLeastOneTodo,
+    isAllTodosCompleted,
+    activeTodosCount,
+    isThereAtLeastOneCompletedTodo,
+  } = useMemo(() => {
+    return {
+      isThereAlLeastOneTodo: !!todos.length,
+      isAllTodosCompleted: todos.every(todo => todo.completed),
+      activeTodosCount: todos.filter(todo => !todo.completed).length,
+      isThereAtLeastOneCompletedTodo: todos.some(todo => todo.completed),
+    };
+  }, [todos]);
+
+  const isTodoLoading = useCallback(
```

Comment:
```suggestion
  const isTodoLoading = 
```
do we need useCallback here?